src/Facebook/FileUpload/FacebookVideoChunk.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright 2014 Facebook, Inc.
+ *
+ * You are hereby granted a non-exclusive, worldwide, royalty-free license to
+ * use, copy, modify, and distribute this software in source code or binary
+ * form for use in connection with the web services and APIs provided by
+ * Facebook.
+ *
+ * As with any software that integrates with the Facebook platform, your use
+ * of this software is subject to the Facebook Developer Principles and
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\FileUpload;
+
+use Facebook\Exceptions\FacebookSDKException;
+
+/**
+ * Class FacebookVideoChunk
+ *
+ * @package Facebook
+ */
+class FacebookVideoChunk extends FacebookVideo
+{
+    /**
+     * @var int The maximum bytes to read
+     */
+    protected $maxlen = -1;
+
+    /**
+     * @var int Seek to the specified start offset before reading.
+     */
+    protected $startOffset = -1;
+
+
+    /**
+     * Creates a new FacebookVideoChunk entity.
+     *
+     * @param string $filePath
+     * @param int|null $maxlen
+     * @param int|null $startOffset
+     *
+     * @throws FacebookSDKException
+     */
+    public function __construct($filePath, $maxlen = null, $startOffset = null)
+    {
+        parent::__construct($filePath);
+        $this->maxlen = isset($maxlen) ? $maxlen : -1;
+        $this->startOffset = isset($startOffset) ? $startOffset : -1;
+    }
+
+    /**
+     * @param int $startOffset
+     */
+    public function setStartOffset($startOffset)
```

Comment:
That's cool - I'm just saying that every time you update a value you return a new instance of it so all instances are immutable. Just keeps us from depending on a global/shared state. :)